src/uhs/transaction/wallet.cpp

```diff
@@ -121,6 +154,35 @@ namespace cbdc {
         return ret;
     }
 
+    auto transaction::wallet::generate_minter_key() -> pubkey_t {
+        if(m_pubkeys.empty()) {
+            // new wallet. create the minter key
+            return generate_key();
```

Comment:
This looks a little odd to me (it seems like I'm missing something). If there are no public keys, `generate_minter_key()` will generate a new keypair (that seems reasonable), but the comment suggests that calling `generate_key()` will ensure that `m_pubkeys[0]` holds the minting key. As far as I can tell, `generate_key()` will return a new key created rather than the minting key, and it doesn't include any calls to functions that would generate the minting key or ensure that `m_pubkeys[0]` is the minting key.

Reply:
@HalosGhost  `m_pubkey[0]` is always the minting key.  If there are no `pubkeys` then `generate_key()` creates one and inserts it at `m_pubkeys[0]` (this is the minting key), otherwise, we use the existing minting key at `m_pubkey[0]`.

Reply:
An offline conversation finally helped this click for me. The first key is always *treated* as the minting key from the wallet's perspective; it's then up to someone configuring the system to specify the correct public key in the configured minting keys.

For the sake of testing determinism, this PR also includes some small machinery to set the minting key for a wallet to be a pre-defined key (a nothing-up-my-sleeve key); but in any tests or real configuration, that should not be used.

I'm all good from this perspective.